gen3/tools/metadata/discovery.py

```
@@ -35,11 +35,23 @@
 
 
 async def output_expanded_discovery_metadata(
-    auth, endpoint=None, limit=500, use_agg_mds=False
+    auth,
+    endpoint=None,
+    limit=500,
+    use_agg_mds=False,
+    guid_type="discovery_metadata",
+    output_format="tsv",
+    output_filename_suffix="",
 ):
     """
-    fetch discovery metadata from a commons and output to {commons}-discovery-metadata.tsv
+    fetch discovery metadata from a commons and output to {commons}-{guid_type}.tsv or {commons}-{guid_type}.json
     """
+    if output_format != "tsv" and output_format != "json":
+        print(
```

you should use logging.error instead of print

---

and maybe raise an exception instead of empty return

---

updated